src/user/online.ts

```diff
@@ -0,0 +1,63 @@
+import db = require('../database');
+import topics = require('../topics');
+import plugins = require('../plugins');
+import meta = require('../meta');
+
+interface userDataTemplate {
+    status: string;
+    lastonline: string;
+}
+
+interface UserTemplate {
+    uid: string;
+    updateLastOnlineTime: (uid: string) => Promise<void>;
+    setUserField: (uid: string, field: string, value: any) => Promise<void>;
```

Comment:
maybe we can set the value to type `number | string`. Not a big deal

Reply:
fixed